### ISSUE TITLE
Wrong color for abbreviations to invalid commands

### ISSUE THREAD

**Issue by Ravenclaw-OIer** [fish-shell/fish-shell#7418]
## Description 
If an abbr points to a non-existent command, fish wrongly colors it as found (that is blue by default).

If an abbr points to a command that *does* exist, but the command *preceding* it (separated by a semicolon) does not exist, then it is also wrongly colored as found.

Basically, when fish replaces abbrs, it blindly assumes that the commands *on this whole line* exist.

## Screenshots:
In the screenshot below, `chtholly` is a nonexistent command, the second invocation of `chtholly` came from replacing `2`, which is an abbr I've created right before that command

![image.png](https://i.loli.net/2020/10/24/FlNaZh3OC97YEgM.png)

## Basic Info
```
fish --version: 3.1.0
uname -a:Linux Ubuntu-Chtholly 5.4.0-52-generic #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
fish-modifications: None
```

**Comment by Ravenclaw-OIer** [comment]
asciinema link: https://asciinema.org/a/keD9rwTcZ9S13SGCBIkLbRMj4

**Comment by zanchey** [comment]
I think this is a duplicate of #5912?

**Comment by Ravenclaw-OIer** [comment]
I did not connect to a remote though, probably different.

**Comment by mqudsi** [comment]
Nice find. I think it's indeed different from #5912, and has to do with the expansion of abbreviations taking place after (or otherwise bypassing) the normal validation routines.

**Comment by ridiculousfish** [comment]
This looks like #5912, except that abbreviations make the problem more obvious / easier to repro because they are expanded right before execution. The plan in that issue (syntax highlight with timeout) still makes sense to me.

**Comment by faho** [comment]
Duplicate of #5912

**Comment by faho** [comment]
(of course if #5912 is fixed, we should check if it works with abbreviations as well, and if it doesn't reopen this one)

**Comment by ridiculousfish** [comment]
I believe this is solidly fixed with d3192d37a2fdf4429c24a29afcb0aad6fe67dbc9
